src/components/ComboBox/ComboBox.tsx

```diff
@@ -0,0 +1,132 @@
+import * as React from 'react';
+import { themr, ThemedComponentClass } from 'react-css-themr';
+import { createUniqueIDFactory } from '@shopify/javascript-utilities/other';
+import { COMBOBOX } from '../ThemeIdentifiers';
+import ComboBoxItem from './ComboBoxItem';
+import Icon from '../Icon';
+import TextField from '../TextField';
+import Popover from '../Popover';
+import arrowSvg from './icons/arrow.svg';
+import * as baseTheme from './ComboBox.scss';
+
+export type Mode = 'collapsible' | 'multiple';
+
+export interface ComboBoxProps {
+  type? : string;
+  key?: string;
+  value: any;
+}
+
+export interface Props {
+  items: ComboBoxProps[];
+  label: string;
+  style?:any;
+}
+
+interface State {
+  open: boolean;
+  items: ComboBoxProps[];
+  initialItems: ComboBoxProps[];
+  anchorEl?: HTMLElement | null;
+  value: string;
+}
+
+class ComboBox extends React.Component<Props, State> {
+  private getUniqueID = createUniqueIDFactory('ComboBox');
+  private id = this.getUniqueID();
+
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      open: false,
+      items: props.items,
+      initialItems: JSON.parse(JSON.stringify(props.items)),
+      value: '',
+    };
+  }
+
+  onChange = (value: string) => {
```

Comment:
@ankitsaraogi Add comments to explain what's happening in this code.  It's a bit complicated.